src/hyperion/external_interaction/callbacks/ispyb_callback_base.py

```diff
@@ -77,41 +91,83 @@ def activity_gated_event(self, doc: Event) -> Event:
-            self.params.hyperion_params.ispyb_params.undulator_gap = doc["data"][
+            self._event_driven_data_collection_info.undulator_gap1 = doc["data"][
                 "undulator_current_gap"
             ]
-            self.params.hyperion_params.ispyb_params.synchrotron_mode = doc["data"][
+            self._event_driven_data_collection_info.synchrotron_mode = doc["data"][
                 "synchrotron-synchrotron_mode"
             ]
-            self.params.hyperion_params.ispyb_params.slit_gap_size_x = doc["data"][
+            self._event_driven_data_collection_info.slitgap_horizontal = doc["data"][
                 "s4_slit_gaps_xgap"
             ]
-            self.params.hyperion_params.ispyb_params.slit_gap_size_y = doc["data"][
+            self._event_driven_data_collection_info.slitgap_vertical = doc["data"][
                 "s4_slit_gaps_ygap"
             ]
-            self.params.hyperion_params.ispyb_params.sample_barcode = doc["data"][
-                "robot-barcode"
-            ]
+            self._sample_barcode = doc["data"]["robot-barcode"]
 
         if event_descriptor.get("name") == CONST.PLAN.ISPYB_TRANSMISSION_FLUX_READ:
-            self.params.hyperion_params.ispyb_params.transmission_fraction = doc[
-                "data"
-            ]["attenuator_actual_transmission"]
-            self.params.hyperion_params.ispyb_params.flux = doc["data"][
+            assert self._event_driven_data_collection_info
+            if doc["data"]["attenuator_actual_transmission"]:
+                # Ispyb wants the transmission in a percentage, we use fractions
+                self._event_driven_data_collection_info.transmission = (
+                    doc["data"]["attenuator_actual_transmission"] * 100
+                )
+                # TODO 1173 Remove this once nexus_utils no longer needs it
+                self.params.hyperion_params.ispyb_params.transmission_fraction = doc[
+                    "data"
+                ]["attenuator_actual_transmission"]
+            self._event_driven_data_collection_info.flux = doc["data"][
                 "flux_flux_reading"
             ]
-            self.params.hyperion_params.ispyb_params.current_energy_ev = (
-                doc["data"]["dcm_energy_in_kev"] * 1000
+            # TODO 1173 Remove this once nexus_utils no longer needs it
+            self.params.hyperion_params.ispyb_params.flux = (
+                self._event_driven_data_collection_info.flux
+            )
+            if doc["data"]["dcm_energy_in_kev"]:
+                energy_ev = doc["data"]["dcm_energy_in_kev"] * 1000
+                self._event_driven_data_collection_info.wavelength = (
+                    convert_eV_to_angstrom(energy_ev)
+                )
+                # TODO 1173 Remove this once nexus_utils no longer needs wavelength_angstroms
+                self.params.hyperion_params.ispyb_params.current_energy_ev = energy_ev
```

Comment:
Created https://github.com/DiamondLightSource/hyperion/issues/1292 to remove this in a new PR